src/MassTransit/DependencyInjection/DependencyInjection/InternalScopeExtensions.cs

```diff
@@ -6,9 +6,9 @@ namespace MassTransit.DependencyInjection
 
     static class InternalScopeExtensions
     {
-        public static IDisposable SetCurrentConsumeContext(this IServiceScope scope, ConsumeContext context)
+        public static IDisposable SetCurrentConsumeContext(this IServiceScope scope, ISetScopedConsumeContext setter, ConsumeContext context)
         {
-            return scope.ServiceProvider.GetRequiredService<ScopedConsumeContextProvider>().PushContext(context);
+            return setter.PushContext(scope.ServiceProvider, context);
```

Comment:
Can this extension method be removed, and the argument type for `ISetScopedConsumeContext` be changed to `IServiceScope` ?